functions/billing/index.js

```diff
@@ -57,9 +57,22 @@ exports.stopBilling = async (pubsubEvent, context) => {
   if (pubsubData.costAmount <= pubsubData.budgetAmount) {
     return `No action necessary. (Current cost: ${pubsubData.costAmount})`;
   }
+  
+  if (!PROJECT_ID) {
+    return `No project specified`;
+  }
 
   await _setAuthCredential();
-  if (await _isBillingEnabled(PROJECT_NAME)) {
+  let billingEnabled;
+  try {
+    billingEnabled = await _isBillingEnabled(PROJECT_NAME);
+  }
+  catch (e) {
+    billingEnabled = true;
+    console.log('Unable to determine if billing is enabled on specified project, assuming true');
+  }
+  
+  if (billingEnabled) {
     return _disableBillingForProject(PROJECT_NAME);
   } else {
     return 'Billing already disabled';
```

Comment:
Logic could be simplified:

```js
try {
  if (await _isBillingEnabled(PROJECT_NAME)) {
    return _disableBillingForProject(PROJECT_NAME);
  } else {
    return 'Billing already disabled';
  }
} catch (e) {
  console.log('Unable to determine if billing is enabled on specified project, assuming true');
}
```

Reply:
I believe that's slightly different logic, as I wanted to make sure _disableBillingForProject would still be called even if _isBillingEnabled threw an error. I thought about doing a try/catch/finally but figured this would be easier to read. Thoughts?

Reply:
Looking further, I would assume that the try/catch should be within the `_isBillingEnabled` function.

It should either return `true`/`false` and never through an error.

Depends on if the sample wants catch API errors or not.

Reply:
Makes sense, I'll move the try/catch.